src/client/activation/extensionSurvey.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, injectable, optional } from 'inversify';
+import { IApplicationShell } from '../common/application/types';
+import '../common/extensions';
+import { traceDecorators } from '../common/logger';
+import {
+    IBrowserService, IPersistentStateFactory, IRandom
+} from '../common/types';
+import { Common, ExtensionSurveyBanner, LanguageService } from '../common/utils/localize';
+import { sendTelemetryEvent } from '../telemetry';
+import { EventName } from '../telemetry/constants';
+import { IExtensionSurvey } from './types';
+
+// persistent state names, exported to make use of in testing
+export enum extensionSurveyStateKeys {
+    doNotShowAgain = 'doNotShowExtensionSurveyAgain',
+    disableSurveyForTime = 'doNotShowExtensionSurveyUntilTime'
+}
+
+export const timeToDisableSurveyFor = 1000 * 60 * 60 * 24 * 7 * 12; // 12 weeks
+const waitTimeToShowSurvey = 1000 * 60 * 60 * 3; // 3 hours
+
+@injectable()
+export class ExtensionSurveyPrompt implements IExtensionSurvey {
+    constructor(
+        @inject(IApplicationShell) private appShell: IApplicationShell,
+        @inject(IBrowserService) private browserService: IBrowserService,
+        @inject(IPersistentStateFactory) private persistentState: IPersistentStateFactory,
+        @inject(IRandom) private random: IRandom,
+        @optional() private sampleSizePerOneHundredUsers: number = 10,
+        @optional() private _waitTimeToShowSurvey: number = waitTimeToShowSurvey) { }
```

Comment:
```suggestion
        @optional() private waitTimeToShowSurvey: number = waitTimeToShowSurvey) { }
```

Please do not use `_` for private properties (do so  only if a property already exists with the same name)